cypress/e2e/homepage.spec.ts

```diff
@@ -0,0 +1,195 @@
+
+      // Act
+      // NOTE: We assume that there are 3 highlights at the start
+      // as this is the initial state post-reset.
+      // NOTE: We require this because it clicks on the center by default
+      // which is where the text is.
+      // As we search for grandparent element to trigger the state change,
+      // clicking on the text will have the wrong grandparent element
+      // leading to no state change.
+      cy.get("@addHighlightButton").click("top")
+
+      // Assert
+      cy.get("@addHighlightButton").should("be.disabled")
+    })
+
+    it("should remove the button in preview when the hero button content is not given", () => {
+      // Arrange
+      const MOCK_HERO_BUTTON_TEXT = "This is a mock hero button text"
+      toggleSectionDropdown("Hero section")
+      cy.get(HERO_BUTTON_PREVIEW_SELECTOR)
+        .as("heroButtonPreview")
+        .should("exist")
+      cy.get(HERO_BUTTON_INPUT_SELECTOR).clear().blur()
+      cy.get("@heroButtonPreview").should("not.exist")
+
+      // Act
+      cy.get(HERO_BUTTON_INPUT_SELECTOR).type(MOCK_HERO_BUTTON_TEXT).blur()
+
+      // Assert
+      cy.get("@heroButtonPreview")
+        .should("exist")
+        .contains(MOCK_HERO_BUTTON_TEXT)
+    })
+
+    it("should preserve content when swapping between highlights and dropdown for hero section", () => {
+      // Arrange
+      const UPDATED_HERO_BUTTON_TEXT = "This is changed blah blah"
+      toggleSectionDropdown("Hero section")
+
+      // Act
+      cy.get(HERO_SECTION_HIGHLIGHTS_SELECTOR).check()
+      cy.get(HERO_BUTTON_INPUT_SELECTOR).clear().type(UPDATED_HERO_BUTTON_TEXT)
+      cy.get(HERO_SECTION_DROPDOWN_SELECTOR).check()
+      cy.get(HERO_SECTION_HIGHLIGHTS_SELECTOR).check()
+
+      // Assert
+      cy.get(HERO_BUTTON_INPUT_SELECTOR).should(
+        "have.value",
+        UPDATED_HERO_BUTTON_TEXT
+      )
+    })
+
+    it("should limit users to a single resource component", () => {
+      // Arrange
+      // NOTE: Precondition here is that there are NO existing resources
+      cy.get(ADD_SECTION_DROPDOWN_SELECTOR).select(
+        SECTION_DROPDOWN_OPTIONS.DEFAULT
+      )
+      cy.contains("div", "Add a new section")
+        .find("button")
+        .as("addSectionButton")
+        .should("be.disabled")
+
+      // Act
+      cy.get(ADD_SECTION_DROPDOWN_SELECTOR).select(
+        SECTION_DROPDOWN_OPTIONS.RESOURCES
+      )
+      cy.get("@addSectionButton").click()
+
+      // Assert
+      // NOTE: This is an existing bug on `production`
+      // where adding a resource section doesn't disable.
+      cy.get("@addSectionButton").should("be.disabled")
+    })
+
+    it.skip("should snap preview to that section when clicking on the section in the editor", () => {
+      // NOTE: Not testing this because cypress doesn't support viewports checking
+    })
+
+    it("should rearrange the blocks correctly and the preview should also reflect the new order", () => {
```

Comment:
![Screenshot 2023-08-02 at 10 37 28 AM](https://github.com/isomerpages/isomercms-frontend/assets/5507822/da156b63-c6c3-4ed1-9143-aeef359efed6)

Somehow I can't get the last test to pass, it keeps failing after the click

Reply:
Oh I think here it's not react beautiful DND so the selector id might be different

Reply:
hmm so do we need to update the selector?

Reply:
Haven't try yet, will let you know.

Reply:
resolving as this was tested against staging (not local)